File: task6/src/main/java/com/kmnvxh222/task6/db/DBInterface.java
package com.kmnvxh222.task6.db;

import com.kmnvxh222.task6.Contact;

import java.util.List;

public interface DBInterface {

    void addContact(Contact contact, DBListener<Contact> listener);

    void deleteContact(String id, DBListener<String> listener);

    void getAllContacts(DBListener<List<Contact>> listener);

    void updateContact(Contact contact, DBListener<Contact> listener);

    void getContactByID(String id, DBListener<List<Contact>> listener);

    void close();

}

File: task6/src/main/java/com/kmnvxh222/task6/db/DBInterface.kt
package com.kmnvxh222.task6.db

import android.database.sqlite.SQLiteDatabase
import com.kmnvxh222.task6.Contacts

interface DBInterface {

    fun addContact(contact: Contacts, db: SQLiteDatabase?)
    fun deleteContact(id: String, db: SQLiteDatabase?)
    fun getAllContacts(db: SQLiteDatabase?): MutableList<Contacts>
    fun updateContact(contact: Contacts, db: SQLiteDatabase?)
    fun getContactByID(id: String, db: SQLiteDatabase?): Contacts?

}
